### DISCUSSION TITLE
Authenticate via GH App using SSH keys

### DISCUSSION THREAD

**Discussion by githubcdr** [renovatebot/renovate#18827] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest version

### If you're self-hosting Renovate, select which platform you are using.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

Github

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

We are having difficulties configuring Renovate using SSH private key authentication using Github Apps.

It seems support for "GITHUB_APP_KEY" has been removed and this got replaced with PAT based authentication. We would like to use GH Apps authentication to prevent API limit errors.

- Is there a way to setup Renovate to use SSH key authentication via GH apps?
- Was this option removed or has it been changed by name?
- Previous versions of renovate would import "renovate.private-key.pem", has this been removed?
- https://github.com/mend/renovate-on-prem/blob/main/docs/configuration-github.md#configure-renovate-pro refers to GITHUB_APP_KEY

Thanks for your support!

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
GITHUB_APP_KEY has never been an option in this repo to my knowledge, e.g. https://github.com/renovatebot/renovate/search?q=GITHUB_APP_KEY

SSH authentication is also *impossible* for GitHub Apps, because GitHub apps only get assigned the ability to generate tokens.

Accordingly I have converted this to a discussion, because there seems to be a misunderstanding

**Reply by githubcdr** [reply]
Hi, I'm sorry but I mixed-up the Enterprise version of renovate with the opensource version. We'd like to migrate to the opensource version but it was unclear to us that this version has no support for SSH based authentication.

The issue for us is that the Github API limit will be reached too soon with just PAT based authentication

**Reply by viceice** [reply]
renovate on-premise has also no ssh based authentication. so you'r doing something wrong.

how can you get rate limits when the hosted renovate app works with a single github app key and doesn't get limited?

**Reply by calexandre** [reply]
I'm also a bit confused on how to setup renovate using GitHub App authentication.
For example, where should we set the github app private-key pem file option?
